org-ql.el

```diff
@@ -448,6 +452,29 @@ Returns cons (INHERITED-TAGS . LOCAL-TAGS)."
                  org-ql-tags-cache))
       (puthash position all-tags tags-cache))))
 
+(defun org-ql--expand-tag-groups (tags &optional exclude-groups)
+  (if (eq tags 'org-ql-nil)
+      'org-ql-nil
+    (let ((exclude-groups (append tags exclude-groups))
+          result)
+      (let (group-tags)
+        (dolist (tag tags)
+          (dolist (groups org-tag-groups-alist)
+            (when (and (not (cl-member (car groups) exclude-groups
```

Comment:
Plain `member `would probably be sufficient here.  (Theoretically, `cl-member` calling `string=` directly might be better, but since `member` is a C subr, I'm guessing it might actually be faster to call `member`, even though it calls a non-type-specific test function, `equal`.)

Reply:
You are absolutely right. Addressed in the new commit.